shared/constants/network.ts

```diff
@@ -0,0 +1,579 @@
+import { capitalize } from 'lodash';
+
+/**
+ * Throughout the extension we set the current provider by referencing its
+ * "type", which can be any of the values in the below object. These values
+ * represent the built-in networks of MetaMask, including test nets, as well
+ * as "rpc" which is the "type" of a custom network added by the user or via
+ * wallet_addEthereumChain.
+ */
+export const NETWORK_TYPES = {
+  GOERLI: 'goerli',
+  KOVAN: 'kovan',
+  LOCALHOST: 'localhost',
+  MAINNET: 'mainnet',
+  RINKEBY: 'rinkeby',
+  ROPSTEN: 'ropsten',
+  RPC: 'rpc',
+} as const;
+
+/**
+ * An object containing shortcut names for any non-builtin network. We need
+ * this to be able to differentiate between networks that require custom
+ * sections of code for our various features, such as swaps or token lists.
+ */
+export const NETWORK_NAMES = {
+  HOMESTEAD: 'homestead',
+};
+
+/**
+ * A type representing any valid value for 'type' for setProviderType and other
+ * methods that add or manipulate networks in MetaMask state.
+ */
+export type NetworkTypes = typeof NETWORK_TYPES[keyof typeof NETWORK_TYPES];
```

Comment:
A couple of guidelines that I've been following is to use singular names for union types and to organize files so that types are the top so they are easier to find. What do you think about these?

Reply:
Does it matter if the type uses the value to create the type definition? I feel thats use before declaration but maybe that doesn't matter for types. I feel in a lot of cases separating the type and the value by a large amount of space leads to confusion but I could be swayed.

Reply:
The compiler won't have a problem with it, but if it's more readable to put the `const` first in this case, then I can understand why you'd want to keep it together.

Reply:
Okay, aside from that, maybe use a singular for this? i.e. `NetworkType` instead of `NetworkTypes`. Otherwise it could be interpreted as a tuple of network types, e.g.

``` typescript
export type NetworkTypes = ['mainnet', 'ropsten', 'goerli', ...]
``` 

as opposed to a union:

``` typescript
export type NetworkType = 'mainnet' | 'ropsten' | 'goerli' | ...
```

Reply:
I moved the types to the top and made the plural -> singular change. I do like that the first part of the file looks like documentation and the rest looks like code. Lets standardize this going forward until we hit a case where it doesn't work.